File: Leetcode/Single Number/Single Number.java
// Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
class Solution {
    public int singleNumber(int[] nums) {
    //using HashMap

        // Initializing a hashmap
        HashMap<Integer,Integer> hm =  new HashMap<Integer,Integer>();
        // Creating a hashmap for given array
        for(int i : nums){
            if(hm.containsKey(i)){
                hm.put(i,hm.get(i) + 1);
            }
            else{
                hm.put(i,1);
            }
        }
        for(int i : nums){
            // if elements occurs only once then return that element
            if(hm.get(i) == 1){
                return i;
            }
        }
        return 0;
    }
}

File: Leetcode/Single Number/Single Number.c
// Given a non-empty array of integers nums, every element appears twice except for one. Find that single one.
int singleNumber(int* nums, int numsSize) {
    //initialize result to 0 to avoid garbage value
    int i, result = 0, size = numsSize;
    //iterate through array
    for (i = 0; i < size; i++) {
        //take xor of every number, xor of two same number results in 0 and hence the final result will be one occuring only once
        result ^= nums[i];

    }
	return result;
}